testing/integration/broker/broker_test.go

```diff
@@ -32,191 +24,200 @@ const (
-		}, PollingTimeout, 5*time.Second).Should(HaveLastOperationState(domain.Succeeded))
+		By("fetching the binding credentials", func() {
+			res := broker.GetBinding(
+				instanceID,
+				bindingID,
+				provisionValues.ServiceID,
+				provisionValues.PlanID,
+			)
+			Expect(res.Code).To(Equal(http.StatusOK))
+			err := json.NewDecoder(res.Result().Body).Decode(&binding)
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(binding.Credentials.AWSAccessKeyID).ToNot(BeEmpty())
+			Expect(binding.Credentials.AWSSecretAccessKey).ToNot(BeEmpty())
+			Expect(binding.Credentials.AWSRegion).ToNot(BeEmpty())
+			Expect(binding.Credentials.PrimaryQueueURL).ToNot(BeEmpty())
+			Expect(binding.Credentials.SecondaryQueueURL).ToNot(BeEmpty())
+		})
 
+		By("using binding credentials to access the service", func() {
+			sess := session.Must(session.NewSession(&aws.Config{
+				Region: aws.String(binding.Credentials.AWSRegion),
+				Credentials: credentials.NewStaticCredentials(
+					binding.Credentials.AWSAccessKeyID,
+					binding.Credentials.AWSSecretAccessKey,
+					"",
+				),
+			}))
+			sqsClient := awssqs.New(sess)
+
+			_, err := sqsClient.SendMessage(&awssqs.SendMessageInput{
+				MessageBody: aws.String("Hello World."),
+				QueueUrl:    aws.String(binding.Credentials.PrimaryQueueURL),
+			})
+			Expect(err).ToNot(HaveOccurred())
+
+			_, err = sqsClient.ReceiveMessage(&awssqs.ReceiveMessageInput{
+				QueueUrl:            aws.String(binding.Credentials.PrimaryQueueURL),
+				MaxNumberOfMessages: aws.Int64(10),
+			})
+			Expect(err).ToNot(HaveOccurred())
+		})
 	})
 
 })
 
-func DeprovisionService(brokerTester brokertesting.BrokerTester, instanceID, serviceID, planID string) {
-	By("Deprovisioning")
-	res := brokerTester.Deprovision(instanceID, serviceID, planID, true)
-	Expect(res.Code).To(Equal(http.StatusAccepted))
-
-	By("waiting for the deprovision to succeed")
-	Eventually(func() *httptest.ResponseRecorder {
-		return brokerTester.LastOperation(instanceID, serviceID, planID, sqs.DeprovisionOperation)
-	}, PollingTimeout, 5*time.Second).Should(HaveLastOperationState(domain.Succeeded))
-}
-
-func Unbind(brokerTester brokertesting.BrokerTester, instanceID string, serviceID string, planID string, bindingID string) {
-	By(fmt.Sprintf("Deferred: Unbinding the %s binding", bindingID))
-	res := brokerTester.Unbind(instanceID, serviceID, planID, bindingID, true)
-	Expect(res.Code).To(Equal(http.StatusAccepted))
-
-	By("waiting for the unbind to complete")
-	Eventually(func() *httptest.ResponseRecorder {
-		return brokerTester.LastBindingOperation(instanceID, bindingID, serviceID, planID, sqs.UnbindOperation)
-	}, PollingTimeout, 5*time.Second).Should(HaveLastOperationState(domain.Succeeded))
+// lastServiceOperationChan returns a channel that contains the latest provision
+// state each time a value is pulled off the channel, binding operated gets
+// polled and queues up the next state. this makes writing assertions cleaner
```

Comment:
I tried to reword it, but I'm not sure I wrote it well

```suggestion
// lastServiceOperationChan returns a channel that repeatedly receives the latest provision
// state.  Each time a value is pulled off the channel, the "last service operation" endpoint gets
// polled and sent to the channel.  This makes writing assertions cleaner.
```